src/basic_types.jl

```diff
@@ -77,19 +77,22 @@ struct BoundaryConditionDoNothing end
 @inline function (::BoundaryConditionDoNothing)(
     u_inner, orientation_or_normal_direction, direction::Integer, x, t, surface_flux, equations)
   return flux(u_inner, orientation_or_normal_direction, equations)
+  # TODO: should this be switched to `surface_flux(u_inner, u_inner, orientation_or_normal_direction, equations)` for consistency?
 end
 
 # This version can be called by hyperbolic solvers on unstructured, curved meshes
 @inline function (::BoundaryConditionDoNothing)(
     u_inner, outward_direction::AbstractVector, x, t, surface_flux, equations)
-  return flux(u_inner, outward_direction, equations)
+
+  # this should reduce to `flux(u_inner, outward_direction, equations)` for a consistent symmetric flux.
+  return surface_flux(u_inner, u_inner, outward_direction, equations)
```

Comment:
This change was necessary to deal with non-conservative terms. 

If the surface flux is consistent such that 
```julia
surface_flux(u_inner, u_inner, outward_direction, equations) = flux(u_inner, outward_direction, equations)
```
then this shouldn't be a breaking change.

Reply:
Sure - for consistency we evaluate both the classical and non-symmetric parts of the flux when evaluating `surface_flux` on the boundary (see for example https://github.com/trixi-framework/Trixi.jl/blob/7dd423222aa800c635bbdbdcb3c3211736e3d601/src/solvers/dgsem_tree/dg_2d.jl#L703-L736) 

We can evaluate `flux` instead of `surface_flux` for equations with only conservative terms, but there is no `flux` for non-conservative terms. Thus, I evaluated `surface_flux` instead, which can be done for both the symmetric and non-symmetric fluxes. 

FWIW, this will probably be overwritten when we rework `BoundaryConditionDirichlet` to dispatch on `have_nonconservative_terms`.